BuildPropertiesPlugin/plugin/src/main/groovy/com/novoda/buildproperties/BuildPropertiesPlugin.groovy

```diff
@@ -0,0 +1,170 @@
+                target.buildConfigField 'String', name ?: formatBuildConfigField(key), "\"${properties[key]}\""
+            }
+        }
+        target.ext.buildConfigProperties = { Closure<BuildProperties> getBuildProperties ->
+            project.afterEvaluate {
+                Properties properties = getBuildProperties().entries
+                properties.stringPropertyNames().each { name ->
+                    target.buildConfigField 'String', formatBuildConfigField(name), "\"${properties[name]}\""
+                }
+            }
+        }
+    }
+
+    private String formatBuildConfigField(String name) {
+        return splitTokens(name)
+                .toUpperCase()
+    }
+
+    private String splitTokens(String name) {
+        return name
+                .replace('.', '_')
+                .split("(?<!(^|[A-Z]))(?=[A-Z])|(?<!^)(?=[A-Z][a-z])")
+                .join('_')
+    }
+
+    private void addResValueSupportTo(target, Project project) {
+        def resValue = { String type, String name, Closure<String> value ->
+            target.resValue type, name, value()
+        }
+        target.ext.resValueBoolean = { String name, Boolean value ->
+            resValue('bool', name, { Boolean.toString(value) })
+        }
+        target.ext.resValueInt = { String name, Integer value ->
+            resValue('integer', name, { Integer.toString(value) })
+        }
+        target.ext.resValueString = { String name, String value ->
+            resValue('string', name, { "\"$value\"" })
+        }
+        target.ext.resValueProperty = { String name = null, Closure<BuildProperties> getBuildProperties, String key ->
+            project.afterEvaluate {
+                Properties properties = getBuildProperties().entries
+                target.resValue 'string', name ?: formatResValueName(key), "\"${properties[key]}\""
+            }
+        }
+        target.ext.resValueProperties = { Closure<BuildProperties> getBuildProperties ->
+            project.afterEvaluate {
+                Properties properties = getBuildProperties().entries
+                properties.stringPropertyNames().each { name ->
+                    target.resValue 'string', formatResValueName(name), properties.getProperty(name)
+                }
+            }
+        }
+    }
+
+    private String formatResValueName(String name) {
+        return splitTokens(name)
+                .toLowerCase()
+    }
+
+    private void addSigningConfigSupportTo(target, Project project) {
+        target.ext.signingConfigProperties = { Closure<BuildProperties> getBuildProperties ->
+            project.afterEvaluate {
+                BuildProperties buildProperties = getBuildProperties()
+                Properties properties = buildProperties.entries
+                target.storeFile new File(buildProperties.parentFile, properties.storeFile)
+                target.storePassword properties.storePassword
```

Comment:
The property file needs to have these values, right. Is it obvious or should we have check, or should we have explanation in the readme?